### PR TITLE
refactor(rating): update to latest conventions

### PR DESCRIPTION
**Pull request by Foxandxss**
This component still needs sweet love, but let put it pretty first.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
LGTM

File: src/rating/rating.ts [hunk 1]
```diff
@@ -13,26 +13,18 @@ import {Component, Input, Output, NgFor, EventEmitter, OnInit} from 'angular2/an
   directives: [NgFor]
 })
 export class NgbRating implements OnInit {
-  @Input() private max: number = 10;
-  @Input() private rate: number;
-  @Input() private readonly: boolean;
-  @Output() private rateChange: EventEmitter = new EventEmitter();
-  @Output() private hover: EventEmitter = new EventEmitter();
-  @Output() private leave: EventEmitter = new EventEmitter();
-  private oldRate: number;
```
Comment (wesleycho) [inline src/rating/rating.ts]: I thought we were using `range: number[]` for arrays?

Comment (Foxandxss) [inline src/rating/rating.ts]: You quoted the wrong line. But we were 2vs1, you and I decided to go Array<foo>; and Pawel decided to go number[].

File: src/rating/rating.ts [hunk 2]
```diff
@@ -44,14 +36,22 @@ export class NgbRating implements OnInit {
...
   update(value: number): void {
     if (!this.readonly) {
-      this.oldRate = value;
+      this._oldRate = value;
       this.rate = value;
       this.rateChange.next(value);
     }
   }
+
+  private _buildTemplateObjects(): Array<number> {
```
Comment (wesleycho) [inline src/rating/rating.ts]: Ditto

File: src/rating/rating.ts (hunk 1 above)
Comment (wesleycho) [inline src/rating/rating.ts]: Bah, I blame GitHub - I was in the split diff view.

**Comment by wesleycho** [comment]
LGTM

**Comment by Foxandxss** [comment]
So what convention you decide ?

**Pull request closed without merging**